modules/imaging_uploader/jsx/UploadForm.js

```diff
@@ -280,67 +272,85 @@ class UploadForm extends Component {
-                 + 'Select this upload in the result table '
-                 + 'to view the processing progress';
+                + 'Select this upload in the result table '
+                + 'to view the processing progress';
         }
         swal.fire({
           title: 'Upload Successful!',
           text: text,
           type: 'success',
         });
         window.location.assign(loris.BaseURL + '/imaging_uploader/');
-      },
-      // Upon errors in upload:
-      // - Displays pop up window with submission error message
-      // - Updates errorMessage and hasError so relevant errors are displayed on form
-      // - Returns to Upload tab
-      error: (error, textStatus, errorThrown) => {
-        let errorMessage = Object.assign({}, this.state.errorMessage);
-        let hasError = Object.assign({}, this.state.hasError);
-        let messageToPrint = '';
-        if (error.responseJSON && error.responseJSON.errors) {
-          errorMessage = error.responseJSON.errors;
-        } else if (error.status == 0) {
-          errorMessage = {
-            'mriFile': ['Upload failed: a network error occured'],
-          };
-        } else if (error.status == 413) {
-          errorMessage = {
-            'mriFile': [
-              'Please make sure files are not larger than '
-              + this.props.maxUploadSize,
-            ],
-          };
-        } else {
-          errorMessage = {
-            'mriFile': [
-              'Upload failed: received HTTP response code '
-              + error.status,
-            ],
-          };
-        }
-        for (let i in errorMessage) {
-          if (errorMessage.hasOwnProperty(i)) {
-            errorMessage[i] = errorMessage[i].toString();
-            if (errorMessage[i].length) {
-              hasError[i] = true;
-              messageToPrint += errorMessage[i] + '\n';
-            } else {
-              hasError[i] = false;
-            }
-          }
-        }
-        swal.fire({
-          title: 'Submission error!',
-          text: messageToPrint,
-          type: 'error',
-        });
-        this.setState({
-          uploadProgress: -1,
-          errorMessage: errorMessage,
-          hasError: hasError,
-        });
-      },
+      } else {
+        this.processError(xhr);
+      }
+    }, false);
+
+    xhr.addEventListener('error', () => {
+      this.processError(xhr);
+    }, false);
+
+    xhr.open('POST', loris.BaseURL + '/imaging_uploader/');
+    xhr.send(formObj);
+  }
+
+  /**
+   * Process XMLHttpRequest errors
+   * @param {XMLHttpRequest} xhr - XMLHttpRequest
+   */
+  processError(xhr) {
+    // Upon errors in upload:
+    // - Displays pop up window with submission error message
+    // - Updates errorMessage and hasError so relevant errors are displayed on form
+    // - Returns to Upload tab
+
+    console.error(xhr.status + ': ' + xhr.statusText);
+
+    let errorMessage = Object.assign({}, this.state.errorMessage);
```

Comment:
This assignment is unused as it will always be reassigned by the if else block below. Just define the variable instead.

Reply:
There is one case where this assignment is used, if xhr.responseis is true and resp.errors is false.